packages/blocks/azure/src/lib/actions/azure-resource-graph-action.ts

```diff
@@ -0,0 +1,243 @@
+  url: string,
+  headers: AxiosHeaders,
+  query: string,
+  batch?: string[],
+  hardLimit?: number,
+): Promise<Record<string, unknown>[]> => {
+  const results: Record<string, unknown>[] = [];
+  let skipToken: string | undefined;
+
+  do {
+    const requestBody = buildRequestBody(query, batch, skipToken);
+    const response = await makeHttpRequest<ResourceGraphResponse>(
+      'POST',
+      url,
+      headers,
+      requestBody,
+    );
+
+    if (response.data?.length) {
+      if (hardLimit) {
+        const remaining = hardLimit - results.length;
+        results.push(...response.data.slice(0, remaining));
+        if (results.length >= hardLimit) {
+          break;
+        }
+      } else {
+        results.push(...response.data);
+      }
+    }
+
+    skipToken = response.$skipToken;
+  } while (skipToken);
+
+  return results;
+};
+
+export const azureResourceGraphAction = createAction({
+  auth: azureAuth,
+  name: 'resource_graph_query',
+  description:
+    'Query Azure Resource Graph using KQL to retrieve resources across multiple subscriptions',
+  displayName: 'Azure Resource Graph Query',
+  isWriteAction: false,
+  props: {
+    useHostSession: getUseHostSessionProperty('Azure', 'az login'),
+    subscriptions: subDropdown,
+    query: Property.LongText({
+      displayName: 'KQL Query',
+      description: 'The Kusto Query Language (KQL) query to execute.',
+      required: true,
+    }),
+    querySubscriptions: Property.Array({
+      displayName: 'Query Subscription IDs',
+      description:
+        'Array of Azure subscription IDs to query. Leave empty to query all accessible subscriptions.',
+      required: false,
+    }),
+    limitSubscriptions: Property.Checkbox({
+      displayName: 'Limit Subscriptions',
+      description:
+        'Enable to limit the number of subscriptions queried. Azure Resource Graph has a limit of 1000 subscriptions per query.',
+      required: false,
+      defaultValue: true,
+    }),
+    subscriptionLimit: Property.DynamicProperties({
+      displayName: '',
+      required: true,
+      refreshers: ['limitSubscriptions'],
+      props: async ({ limitSubscriptions }) => {
+        if (limitSubscriptions) {
+          return {
+            limit: Property.Number({
+              displayName: 'Subscription Limit',
+              description:
+                'Maximum number of subscriptions to include in the query. Default is 1000 (Azure API limit).',
+              required: false,
+              defaultValue: 1000,
+              validators: [Validators.minValue(1), Validators.integer],
+            }),
+          } as any;
+        }
+        return {};
+      },
+    }),
+    maxResults: Property.Number({
+      displayName: 'Maximum Results',
+      description:
+        'Maximum number of results to return. Leave empty for no limit.',
+      required: false,
+      validators: [Validators.minValue(1), Validators.integer],
+    }),
+    apiVersion: Property.ShortText({
```

Comment:
idk if i should have added this or not, i feel like its a pain if we have to change it in the backend everytime theres a new one